compiler warning from MKLDNNLayer and so on

There are several compiler warning:
- MKLDNNConvLayer.cpp
```
[15:04:48]	In file included from /paddle/build/third_party/install/protobuf/include/google/protobuf/stubs/common.h:40:0,
[15:04:48]	from /paddle/build/proto/ModelConfig.pb.h:9,
[15:04:48]	from /paddle/paddle/gserver/layers/Layer.h:19,
[15:04:48]	from /paddle/paddle/gserver/layers/MKLDNNLayer.h:18,
[15:04:48]	from /paddle/paddle/gserver/layers/MKLDNNConvLayer.h:17,
[15:04:48]	from /paddle/paddle/gserver/layers/MKLDNNConvLayer.cpp:15:
[15:04:48]	/paddle/build/third_party/install/glog/include/glog/logging.h: In instantiation of 'std::__cxx11::string* google::Check_EQImpl(const T1&, const T2&, const char*) [with T1 = long unsigned int; T2 = int; std::__cxx11::string = std::__cxx11::basic_string<char>]':
[15:04:48]	/paddle/paddle/gserver/layers/MKLDNNConvLayer.cpp:31:3: required from here
[15:04:48]	/paddle/build/third_party/install/glog/include/glog/logging.h:717:32: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[15:04:48]	DEFINE_CHECK_OP_IMPL(Check_EQ, ==) // Compilation error with CHECK_EQ(NULL, x)?
[15:04:48]	^
[15:04:48]	/paddle/build/third_party/install/protobuf/include/google/protobuf/stubs/port.h:200:53: note: in definition of macro 'GOOGLE_PREDICT_TRUE'
[15:04:48]	#define GOOGLE_PREDICT_TRUE(x) (__builtin_expect(!!(x), 1))
[15:04:48]	^
[15:04:48]	/paddle/build/third_party/install/glog/include/glog/logging.h:717:1: note: in expansion of macro 'DEFINE_CHECK_OP_IMPL'
[15:04:48]	DEFINE_CHECK_OP_IMPL(Check_EQ, ==) // Compilation error with CHECK_EQ(NULL, x)?
[15:04:48]
```
- MKLDNNFcLayer.cpp
```
[15:05:37]	In file included from /paddle/build/third_party/install/protobuf/include/google/protobuf/stubs/common.h:40:0,
[15:05:37]	from /paddle/build/proto/ModelConfig.pb.h:9,
[15:05:37]	from /paddle/paddle/gserver/layers/Layer.h:19,
[15:05:37]	from /paddle/paddle/gserver/layers/MKLDNNLayer.h:18,
[15:05:37]	from /paddle/paddle/gserver/layers/MKLDNNFcLayer.h:17,
[15:05:37]	from /paddle/paddle/gserver/layers/MKLDNNFcLayer.cpp:15:
[15:05:37]	/paddle/build/third_party/install/glog/include/glog/logging.h: In instantiation of 'std::__cxx11::string* google::Check_EQImpl(const T1&, const T2&, const char*) [with T1 = long unsigned int; T2 = int; std::__cxx11::string = std::__cxx11::basic_string<char>]':
[15:05:37]	/paddle/paddle/gserver/layers/MKLDNNFcLayer.cpp:31:3: required from here
[15:05:37]	/paddle/build/third_party/install/glog/include/glog/logging.h:717:32: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[15:05:37]	DEFINE_CHECK_OP_IMPL(Check_EQ, ==) // Compilation error with CHECK_EQ(NULL, x)?
[15:05:37]	^
[15:05:37]	/paddle/build/third_party/install/protobuf/include/google/protobuf/stubs/port.h:200:53: note: in definition of macro 'GOOGLE_PREDICT_TRUE'
[15:05:37]	#define GOOGLE_PREDICT_TRUE(x) (__builtin_expect(!!(x), 1))
[15:05:37]	^
[15:05:37]	/paddle/build/third_party/install/glog/include/glog/logging.h:717:1: note: in expansion of macro 'DEFINE_CHECK_OP_IMPL'
[15:05:37]	DEFINE_CHECK_OP_IMPL(Check_EQ, ==) // Compilation error with CHECK_EQ(NULL, x)?
```
- test_CrossEntropyOverBeamGrad.cpp
```
[15:10:50]	In file included from /paddle/build/third_party/install/protobuf/include/google/protobuf/stubs/common.h:40:0,
[15:10:50]	from /paddle/build/proto/ModelConfig.pb.h:9,
[15:10:50]	from /paddle/paddle/gserver/tests/test_CrossEntropyOverBeamGrad.cpp:19:
[15:10:50]	/paddle/build/third_party/install/glog/include/glog/logging.h: In instantiation of 'std::__cxx11::string* google::Check_LEImpl(const T1&, const T2&, const char*) [with T1 = int; T2 = long unsigned int; std::__cxx11::string = std::__cxx11::basic_string<char>]':
[15:10:50]	/paddle/paddle/gserver/tests/test_CrossEntropyOverBeamGrad.cpp:231:9: required from here
[15:10:50]	/paddle/build/third_party/install/glog/include/glog/logging.h:719:32: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[15:10:50]	DEFINE_CHECK_OP_IMPL(Check_LE, <=)
[15:10:50]	^
[15:10:50]	/paddle/build/third_party/install/protobuf/include/google/protobuf/stubs/port.h:200:53: note: in definition of macro 'GOOGLE_PREDICT_TRUE'
[15:10:50]	#define GOOGLE_PREDICT_TRUE(x) (__builtin_expect(!!(x), 1))
[15:10:50]	^
[15:10:50]	/paddle/build/third_party/install/glog/include/glog/logging.h:719:1: note: in expansion of macro 'DEFINE_CHECK_OP_IMPL'
[15:10:50]	DEFINE_CHECK_OP_IMPL(Check_LE, <=)
[15:10:50]	^
[15:05:37]
```
- lod_tensor_test.cu
```
[15:03:17]	/paddle/build/third_party/install/glog/include/glog/logging.h: In instantiation of 'std::__cxx11::string* google::Check_EQImpl(const T1&, const T2&, const char*) [with T1 = long unsigned int; T2 = int; std::__cxx11::string = std::__cxx11::basic_string<char>]':
[15:03:17]	/paddle/paddle/framework/lod_tensor_test.cu:39:191: required from here
[15:03:17]	/paddle/build/third_party/install/glog/include/glog/logging.h:717:153: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
```

--- a/paddle/framework/lod_tensor_test.cu
+++ b/paddle/framework/lod_tensor_test.cu
@@ -36,8 +36,8 @@ TEST(LoDTensor, LoDInGPU) {
   lod_tensor.mutable_data<float>(place);
 
   lod_tensor.set_lod(src_lod);
-  CHECK_EQ(lod_tensor.lod_element(0, 2), 4);
-  CHECK_EQ(lod_tensor.lod_element(0, 4), 8);
+  CHECK_EQ(lod_tensor.lod_element(0, 2), 4UL);
+  CHECK_EQ(lod_tensor.lod_element(0, 4), 8UL);
 
   auto lod = lod_tensor.lod();
 

--- a/paddle/gserver/layers/MKLDNNConvLayer.cpp
+++ b/paddle/gserver/layers/MKLDNNConvLayer.cpp
@@ -28,7 +28,7 @@ bool MKLDNNConvLayer::init(const LayerMap& layerMap,
   if (!MKLDNNLayer::init(layerMap, parameterMap)) {
     return false;
   }
-  CHECK_EQ(inputLayers_.size(), 1) << "Only support one input layer yet";
+  CHECK_EQ(inputLayers_.size(), 1UL) << "Only support one input layer yet";
   CHECK_EQ(inputLayers_.size(), parameters_.size());
   CHECK(config_.shared_biases()) << "Only support shared biases yet";
 

--- a/paddle/gserver/layers/MKLDNNFcLayer.cpp
+++ b/paddle/gserver/layers/MKLDNNFcLayer.cpp
@@ -28,7 +28,7 @@ bool MKLDNNFcLayer::init(const LayerMap& layerMap,
     return false;
   }
 
-  CHECK_EQ(inputLayers_.size(), 1) << "Only support one input layer yet";
+  CHECK_EQ(inputLayers_.size(), 1UL) << "Only support one input layer yet";
   CHECK_EQ(inputLayers_.size(), parameters_.size());
   CHECK(!parameters_[0]->isSparse()) << "Do not support sparse yet";
 

--- a/paddle/gserver/tests/test_CrossEntropyOverBeamGrad.cpp
+++ b/paddle/gserver/tests/test_CrossEntropyOverBeamGrad.cpp
@@ -228,7 +228,7 @@ void genGroundTruth(vector<SingleBeamExpansion>& beamExpansions,
         curBeam.groundTruth[j] = *(start + n);
         curBeam.inBeam[j] = 1;
       } else {
-        CHECK_LE(curBeam.rowIdxInBeam[j] + 1,
+        CHECK_LE((size_t)curBeam.rowIdxInBeam[j] + 1,
                  curBeam.subSeqStartPos.size() - 1);
         int start = curBeam.subSeqStartPos[curBeam.rowIdxInBeam[j]];
         int end = curBeam.subSeqStartPos[curBeam.rowIdxInBeam[j] + 1];

--- a/paddle/operators/multiplex_op.cu
+++ b/paddle/operators/multiplex_op.cu
@@ -42,7 +42,7 @@ class MultiplexGPUKernel : public framework::OpKernel {
     for (auto i = 0; i < rows; i++) {
       int32_t k = index[i];
       PADDLE_ENFORCE_GE(k, 0, "index must be nonnegative.");
-      PADDLE_ENFORCE_LT(k, ins.size(),
+      PADDLE_ENFORCE_LT((size_t)k, ins.size(),
                         "index exceeds the number of candidate tensors.");
       memory::Copy(place, out->data<T>() + i * cols, place,
                    ins[k]->data<T>() + i * cols, cols * sizeof(T), stream);

